File: zantt-api-backend/src/main/java/com/zantt/exceptions/WellKnownApiException.java
package com.zantt.exceptions;

public class WellKnownApiException extends RuntimeException {
    private final String errorCode;

    public String getErrorCode() {
        return errorCode;
    }

    public WellKnownApiException(String message, String errorCode) {
        super(message);
        assert errorCode != null && !errorCode.isEmpty() && !errorCode.trim().isEmpty() :
                "WellKnownApiException [errorCode] is null or empty";
        this.errorCode = errorCode;
    }
}

File: zantt-api-backend-net/Zantt/Exceptions/WellKnownApiException.cs
using System.Diagnostics;

namespace Zantt.Exceptions;

public class WellKnownApiException : Exception
{
    public string ErrorCode { get; private set; }

    public WellKnownApiException(string message, string errorCode) : base(message)
    {
        Debug.Assert(errorCode != null, "ErrorCode must not null");
        Debug.Assert(!string.IsNullOrWhiteSpace(errorCode), "ErrorCode must not empty (whitespace)");

        ErrorCode = errorCode;
    }
}
